src/server/src/main/java/io/cassandrareaper/storage/MemoryStorageFacade.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright 2014-2017 Spotify AB
+ * Copyright 2016-2019 The Last Pickle Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.storage;
+
+import io.cassandrareaper.core.Cluster;
+import io.cassandrareaper.core.DiagEventSubscription;
+import io.cassandrareaper.core.PercentRepairedMetric;
+import io.cassandrareaper.core.RepairRun;
+import io.cassandrareaper.core.RepairSchedule;
+import io.cassandrareaper.core.RepairSegment;
+import io.cassandrareaper.core.RepairUnit;
+import io.cassandrareaper.core.Snapshot;
+import io.cassandrareaper.resources.view.RepairRunStatus;
+import io.cassandrareaper.resources.view.RepairScheduleStatus;
+import io.cassandrareaper.storage.cluster.MemClusterDao;
+import io.cassandrareaper.storage.events.MemEventsDao;
+import io.cassandrareaper.storage.metrics.MemMetricsDao;
+import io.cassandrareaper.storage.repairrun.MemRepairRunDao;
+import io.cassandrareaper.storage.repairschedule.MemRepairScheduleDao;
+import io.cassandrareaper.storage.repairsegment.MemRepairSegment;
+import io.cassandrareaper.storage.repairunit.MemRepairUnitDao;
+import io.cassandrareaper.storage.snapshot.MemSnapshotDao;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Implements the StorageAPI using transient Java classes.
+ */
+public final class MemoryStorageFacade implements IStorage {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MemoryStorageFacade.class);
+  private final MemRepairSegment memRepairSegment = new MemRepairSegment(this);
```

Comment:
I'm happy to make this change in the next PR if that's OK with you. It will take a bit of work and seems like it should be undertaken in a refactoring week so as not to distract from other work.